server/metric_reporter.go

```diff
@@ -0,0 +1,128 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	enterpriseAPI "github.com/bytebase/bytebase/enterprise/api"
+	"github.com/bytebase/bytebase/plugin/metric/collector"
+	"github.com/bytebase/bytebase/plugin/metric/reporter"
+
+	"go.uber.org/zap"
+)
+
+const (
+	metricSchedulerInterval = time.Duration(1) * time.Hour
+)
+
+// MetricScheduler is the metric scheduler.
+type MetricScheduler struct {
+	l *zap.Logger
+
+	mode common.ReleaseMode
+	// subscription is the pointer to the server.subscription. This can be updated by users so we need to update the identify in each schedule.
+	subscription  *enterpriseAPI.Subscription
+	reporter      reporter.MetricReporter
+	workspaceID   string
+	collectorList []collector.MetricCollector
+}
+
+// NewMetricScheduler creates a new metric scheduler.
+func NewMetricScheduler(logger *zap.Logger, server *Server, workspaceID string) *MetricScheduler {
+	reporter := reporter.NewSegmentReporter(logger, server.profile.SegmentKey, workspaceID)
+	collectorList := []collector.MetricCollector{
+		collector.NewInstanceCollector(logger, server.store),
+		collector.NewIssueCollector(logger, server.store),
+	}
+
+	return &MetricScheduler{
+		l:             logger,
+		mode:          server.profile.Mode,
+		subscription:  &server.subscription,
+		workspaceID:   workspaceID,
+		reporter:      reporter,
+		collectorList: collectorList,
+	}
+}
+
+// Run will run the metric scheduler.
+func (m *MetricScheduler) Run(ctx context.Context, wg *sync.WaitGroup) {
+	ticker := time.NewTicker(metricSchedulerInterval)
+	defer ticker.Stop()
+	defer wg.Done()
+	m.l.Debug(fmt.Sprintf("Metrics scheduler started and will run every %v", metricSchedulerInterval))
+
+	for {
+		if m.mode != common.ReleaseModeProd {
```

Comment:
We may want to disable the runner entirely when mode isn't Prod or profile.Demo is true.

Reply:
Updated